src/trousse/dataset.py

```diff
@@ -99,6 +100,103 @@ def _find_single_column_type(df_col: pd.Series) -> Dict[str, str]:
         return {_COL_NAME_COLUMN: col, _COL_TYPE_COLUMN: "mixed_type_col"}
 
 
+def _convert_to_numeric_mixed_types(col_serie: pd.Series):
+    """
+    Convert 'object'-typed values to numerical when possible.
+
+    This static method analyzes the pandas Series ``col_serie`` looking
+    values that can be interpreted as numbers (even if they are string-typed)
+    (e.g. '2' -> 2). The found numbers are converted to the appropriate
+    numeric type.
+
+    Parameters
+    ----------
+    col_serie : pd.Series
+        Series containing the values that will be analyzed. It will not be
+        modified inplace.
+
+    Returns
+    -------
+    pd.Series
+        New pandas Series where numeric values have been converted to the
+        appropriate numeric type.
+    """
+    numeric_col = col_serie.copy()
+    numeric_col = pd.to_numeric(numeric_col, errors="coerce")
+    numeric_map = numeric_col.notna()
+    numeric_ids = np.where(numeric_map)
+    non_numeric_ids = np.where(np.logical_not(numeric_map))
+    mixed_list = (
```

Comment:
I guess most of the code in this method can be simplified this way: 

```python
numeric_col[numeric_col.notna()].tolist() + col_serie[numeric_col.isna()].tolist()
```

IMHO is more readable. Try it out because is not tested